java/backend.scala

```diff
@@ -70,56 +75,214 @@ class Backend {
 }
 
 object Backend {
+  private[this] var isService: Boolean = false
+  private[this] var gatewayServerSocket: ServerSocket = null
+  private[this] var port: Int = 0
+  private[this] var sessionId: Int = 0
+  
+  private[this] var hc: HiveContext = null
+  
+  private[this] var sessionsMap:Map[Int, Int] = Map()
+  
+  object GatewayOperattions extends Enumeration {
+    val GetPorts, RegisterInstance = Value
+  }
+  
+  def getOrCreateHiveContext(sc: SparkContext): HiveContext = {
+    if (hc == null) {
+      hc = new HiveContext(sc)
+    }
+    
+    hc
+  }
+  
+  def portIsAvailable(port: Int) = {
+    var ss: ServerSocket = null
+    var available = false
+    
+    Try {
```

Comment:
Yes, Try swallow exceptions, that was my intention since we are trying to discover is a port is open.